src/select/select.c

```diff
@@ -157,7 +163,8 @@ static void css__destroy_node_data(struct css_node_data *node_data)
 
 	assert(node_data != NULL);
 
-	if (node_data->bloom != NULL) {
+	if (node_data->bloom != NULL &&
+			node_data->bloom != css__get_empty_bloom()) {
```

Comment:
Nice catch, this would have affected the cleanup of the `css_select_state` on error paths if something failed during selection.